Tests/TestingApps/PaywallValidationTester/PaywallValidationTesterView.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallValidationTesterView.swift
+//
+//  Created by Chris Vasselli on 2025/07/09.
+
+import SwiftUI
+import RevenueCat
+@testable import RevenueCatUI
+
+struct PaywallValidationTesterView: View {
+    var offeringId: String? = ProcessInfo.processInfo.environment["OFFERING_ID"]
+    @State var offerings:[Offering] = []
+
+    var body: some View {
+        TabView {
+            ForEach(offerings, id: \.self) { offering in
+                if offering.id == offeringId || offeringId == nil {
```

Comment:
I've tested running the app setting `offeringId` to `nil`. The UX is a bit confusing, since in the middle of the horizontal scrolling to see the next paywall, the UI flashes and sometimes it results in having scrolled two paywalls at once. Not a big deal, but it would be nice to fix.

In case it helps, it seems that all paywalls get re-rendered by SwiftUI when scrolling the `TabView`

Reply:
I dug into this, and I think it's some weird SwiftUI quirkiness around TabViews and nested ForEach. I tried commenting out different pieces to figure out what's causing that flashing is it scrolls to the next page, and I isolated it to, bizarrely, the ForEach in `ComponentsView.layoutComponents`. If I replace the contents of that foreach with a simple Text, I still get the flickering. If I remove the ForEach and have just a single Text block, it doesn't happen.

There may be some other complicated interaction going on, but it's easier to just switch to using a ScrollView and LazyHStack, which is what I've done in the latest commit..

Reply:
Ohh nice investigation! I would have never imagined that the culprit of this would be deeper 🤯 
Thanks for taking a look and thank you for fixing it! I think the `ScrollView` + `LazyHStack` is a great idea! And it makes it more efficient, as before all the paywalls were rendering at once and every time scrolling happened